src/NLog/Config/LoggingConfigurationParser.cs

```diff
@@ -247,7 +247,7 @@ private static string ExpandFilePathVariables(string internalLogFile)
                     internalLogFile = internalLogFile.Replace(tempDirToken, System.IO.Path.GetTempPath() + System.IO.Path.DirectorySeparatorChar.ToString());
 #if !NETSTANDARD1_3
                 if (ContainsSubStringIgnoreCase(internalLogFile, "${processdir}", out string processDirToken))
-                    internalLogFile = internalLogFile.Replace(processDirToken, System.IO.Path.GetDirectoryName(ProcessIDHelper.Instance.CurrentProcessFilePath) + System.IO.Path.DirectorySeparatorChar.ToString());
+                    internalLogFile = internalLogFile.Replace(processDirToken, System.IO.Path.GetDirectoryName(LogFactory.DefaultAppEnvironment.CurrentProcessFilePath) + System.IO.Path.DirectorySeparatorChar.ToString());
```

Comment:
Should this be `CurrentAppEnvironment`? Or don't we have a LogFactory instance her?

Reply:
Change `ExpandFilePathVariables` from static-method to member-method. And adjusted.